src/SegmentedString.chpl

```diff
@@ -488,6 +492,69 @@ module SegmentedString {
       return computeOnSegments(offsets.a, values.a, SegFunction.StringIsTitle, bool);
     }
 
+    proc idnaEncodeDecode(cmd: string) throws {
+      // select the appropriate file based on the command
+      var procFile: string;
+      // we need to add this dir check for testing functionality
+      var basePath: string = realPath(curDir);
+      var (pathName, dirName) = splitPath(basePath);
+      if "tests" == dirName {
+        basePath = pathName;
+      }
+      select cmd {
+        when "encode" {
+          procFile = "/"+basePath+"/src/exec/ak_encode.py";
+        }
+        when "decode" {
+          procFile = "/"+basePath+"/src/exec/ak_decode.py";
+        }
+        otherwise {
+          throw getErrorWithContext(msg="Invalid encode/decode cmd. %s".format(cmd),
+                      lineNumber=getLineNumber(), 
+                      routineName=getRoutineName(), 
+                      moduleName=getModuleName(), 
+                      errorClass='ValueError');
+        }
+      }
+    
+      ref origVals = this.values.a;
+      ref offs = this.offsets.a;
+      var encodeArr: [0..#this.size] string; 
+      var encodeOffsets: [this.offsets.aD] int;
+      var encodeLengths: [this.offsets.aD] int;
+
+      const lengths = this.getLengths();
+      forall (i, off, len) in zip(0..#this.size, offs, lengths) {
+        var str_entry: string;
+        for b in interpretAsString(origVals, off..#len, borrow=true) {
+          str_entry = str_entry + b;
+        }
+        // use subprocessing to make a call to a python file for the encoding
+        var sub = spawn(["python3", procFile, "-v", str_entry], stdout=pipeStyle.pipe);
```

Comment:
Unfortunately, piping does not work on HPC systems that have to register memory with the network (cray networks, infiniband.) https://github.com/Bears-R-Us/arkouda/issues/556 has more details.

I can never keep the details of this fresh in mind, but I think the workaround is to write your output to a file and then read that file in?

Reply:
Good to know. Thanks. I will look into it.